### PR TITLE
Adding Azure Linux (Mariner 2) AMD64 container

### PR DESCRIPTION
**Pull request by AhmedBM**
## Description

* Added new Mariner amd64 build container

## Checklist

- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [X] I added unit-tests to validate my changes. All unit tests are passing.
- [X] I have merged the latest `main` branch prior to this PR submission.
- [X] I submitted this PR against the `main` branch.

### CODE REVIEW

File: devops/docker/mariner-2-amd64/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    openssl-devel \
+    rapidjson-devel \
+    rpm-build \
+    tar \
+    tmux \
+    tree \
+    unzip \
+    util-linux \
+    wget \
+    zip && \
```
Comment (MariusNi) [inline devops/docker/mariner-2-amd64/Dockerfile:26]: Please make sure that we really need all of these to build OSConfig for Mariner. Some seem a bit curious.

**Review by MariusNi — APPROVED** [review]
:shipit:

File: devops/docker/mariner-2-amd64/Dockerfile (hunk 1 above)
Comment (AhmedBM) [reply devops/docker/mariner-2-amd64/Dockerfile:26]: Yes they are all needed and have also been vetted by MSFT as trusted repos and approved packages

Comment (MariusNi) [reply devops/docker/mariner-2-amd64/Dockerfile:26]: For what is needed unzip (line 23)?

Comment (MariusNi) [reply devops/docker/mariner-2-amd64/Dockerfile:26]: These overall look more like for building Mariner from source, not for running Mariner in a container. Gave above an example (unzip), there are others.

Comment (MariusNi) [reply devops/docker/mariner-2-amd64/Dockerfile:26]: Discussed offline. Thanks Ahmed!

**Review by MariusNi — APPROVED** [review]
:shipit:

**Review by simonjaeger — APPROVED** [review]

**Review by robertschaedler3 — APPROVED** [review]

**Pull request merged**